### PR TITLE
DM-31591: Update logging recommendations

### PR DESCRIPTION
**Pull request by timj**
* Use python logging
* No longer explain pex_logging

### CODE REVIEW

File: stack/logging.rst [hunk 1]
```diff
@@ -2,18 +2,22 @@
 Logging
 #######
 
-This page provides guidance to developers for using logging with :lmod:`lsst.log`  in the Science Pipelines code base.
-For reference documentation on the logging framework refer to the `Doxygen page on logging`_.
+This page provides guidance to developers for using logging in the Science Pipelines code base.
+In general, all logging in Python code should be done with the standard :mod:`logging` package.
+All logging from C++ code should be done using the :lmod:`lsst.log` package and reference documentation on that logging framework can be found on the `Doxygen page on logging`_.
```
Comment (ktlim) [inline stack/logging.rst:7]:
```suggestion
All logging from C++ code should be done using the :lmod:`lsst.log` package; reference documentation on that logging framework can be found at the `Doxygen page on logging`_.
```

File: stack/logging.rst [hunk 2]
```diff
@@ -2,18 +2,22 @@
...
 For an example of configuring the logging framework in pipeline tasks, see the `pipelines.lsst.io page on logging`_.
 
 Developers are encouraged to insert log messages whenever and wherever they might be useful, with appropriate component names and levels.
 
-Whether using :lmod:`lsst.log` or any other logging mechanism, timestamps recorded in logs should use Internet `RFC 3339`_ format, which is sortable and includes the timezone.  See the discussion in `DM-1203`_ for history.
+Application code that uses both Python and C++ logging should include configuration code to forward C++ log messages to the Python :mod:`logging` system.
+This is handled automatically when using, for example, the `pipetask`_ command.
```
Comment (ktlim) [inline stack/logging.rst:13]: Move this up to line 8?  That way you have Python, C++, and Python plus C++ in the same paragraph.

File: stack/logging.rst [hunk 3]
```diff
@@ -22,6 +26,13 @@ Logger Names
 
 Logger names should generally start with the fully qualified name of the module/file containing the logger, without the leading ``lsst.``.
 Some example logger names are ``afw.image.MaskedImage`` and ``meas.algorithms.starSelector``.
+A common Python recommendation is to create the logger name from the module hierarchy automatically:
+
+.. code-block:: python
+
+   log = logging.getLogger(__name__.partition(".")[2])
+
+**(Aside to be removed: moving all the logging support code to utils should allow us to have a helper function for this -- an argument could be made that it's better to include the ``lsst.`` prefix to allow simpler global log configuration. It should also allow a proper Rubin getLogger that can return a thing that supports verbose/trace).**
```
Comment (ktlim) [inline stack/logging.rst:35]: The use of Python logging that can conflict with logging by other packages basically mandates that we use `lsst.`, doesn't it?  Want to RFC that?

File: stack/logging.rst [hunk 4]
```diff
@@ -30,35 +41,47 @@ Logger names use ``.`` as component separators, not ``::``, even in C++.
...
-   lsst.log.info('Some information during normal operation')
-   lsst.log.warn('Here is a warning!')
+   import logging
+   logging.info('Some information during normal operation')
+   logging.warning('Here is a warning!')
 
 The example logs to the default (root) logger.
+By default Python logs warning messages and no other log messages.
+This means that the above example code will only write a single log message.
+Application code is required to configure the logging state for libraries and this is generally done by calling :func:`logging.basicConfig` to set a default logger level and default log format.
```
Comment (ktlim) [inline stack/logging.rst:55]: But we do by default configure to produce INFO messages, so only people writing non-Task code would observe this.

File: stack/logging.rst [hunk 5]
```diff
@@ -93,24 +116,29 @@ For example:
...
+- CRITICAL: for severe errors that may prevent further execution of the component (FATAL is also allowed as an alias).
 - ERROR: for errors that may still allow the execution to continue.
-- WARN: for conditions that may indicate a problem but that allow continued execution.
+- WARNING: for conditions that may indicate a problem but that allow continued execution (WARN is also allowed as an alias).
 - INFO: for information that is of interest during normal execution including production.
 - DEBUG: for information that is of interest to developers but not of interest during production.
+
+In addition there are two additional log levels allowed for specialist pipelines-specific loggers (such as those used for :lclass:`lsst.pipe.base.Task`):
+
+- VERBOSE: for messages of a more detailed nature than would normally be expected to be shown by default but that will not swamp the user in the way that DEBUG messages would.
```
Comment (ktlim) [inline stack/logging.rst:130]: I'd still like to say here "But consider using DEBUG messages with a separate logger name that can be enabled when desired, as described below."

File: stack/logging.rst [hunk 6]
```diff
@@ -93,24 +116,29 @@ For example:
...
 - TRACE: for detailed information when debugging.
```
Comment (ktlim) [inline stack/logging.rst:131]: TRACE is particularly intended for use inside loops.

**Review by ktlim — APPROVED** [review]

File: stack/logging.rst (hunk 3 above)
Comment (timj) [reply stack/logging.rst:35]: I can RFC that and see what happens.

File: stack/logging.rst (hunk 5 above)
Comment (timj) [reply stack/logging.rst:130]: Do you mean the section below talking about `TRACEn.` logger names? Or do you mean the code immediately below here? ie are you advocating a `verbose.lsst.xxx` hierarchy reserved for verbose messages?

Comment (ktlim) [reply stack/logging.rst:134]: Doesn't have to start with `verbose`.  That depends on whether synchronized verbosity across different modules is desirable or just increased verbosity in one place.

File: stack/logging.rst (hunk 3 above)
Comment (timj) [reply stack/logging.rst]: RFC-801

**Pull request merged**